features/topology-map/plugins/org.opennms.features.topology.plugins.topo.pathoutage/src/test/java/org/opennms/features/topology/plugins/topo/pathoutage/PathOutageStatusProviderTest.java

```diff
@@ -0,0 +1,317 @@
+		this.outageDao.save(createOutage(EventConstants.NODE_DOWN_EVENT_UEI, address_1, this.nodeDao.get(1), OnmsSeverity.MINOR));
+		stats = this.testStatuses();
+		for (VertexRef vertex : stats.keySet()) {
+			Assert.assertNotNull(stats.get(vertex));
+			if (vertex.getId().equalsIgnoreCase("1")) {
+				Assert.assertEquals(OnmsSeverity.MINOR.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+			} else {
+				Assert.assertEquals(OnmsSeverity.NORMAL.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+			}
+		}
+
+		// Adding a MAJOR path outage to the same node.
+		// PathOutageStatusProvider should display the alarm with a MAJOR severity level
+		this.outageDao.save(createOutage(EventConstants.NODE_DOWN_EVENT_UEI, address_1, this.nodeDao.get(1), OnmsSeverity.MAJOR));
+		stats = this.testStatuses();
+		for (VertexRef vertex : stats.keySet()) {
+			Assert.assertNotNull(stats.get(vertex));
+			if (vertex.getId().equalsIgnoreCase("1")) {
+				Assert.assertEquals(OnmsSeverity.MAJOR.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+				break;
+			}
+		}
+
+		// Adding several more path outages of different types
+		// PathOutageStatusProvider should display them all, plus the one from the previous test
+		this.outageDao.save(createOutage(EventConstants.NODE_LOST_SERVICE_EVENT_UEI, address_2, this.nodeDao.get(2), OnmsSeverity.MINOR));
+		this.outageDao.save(createOutage(EventConstants.INTERFACE_DOWN_EVENT_UEI, address_3 , this.nodeDao.get(3), OnmsSeverity.MINOR));
+		this.outageDao.save(createOutage(EventConstants.PATH_OUTAGE_EVENT_UEI,address_4, this.nodeDao.get(4), OnmsSeverity.MAJOR));
+		stats = this.testStatuses();
+		for (VertexRef vertex : stats.keySet()) {
+			Assert.assertNotNull(stats.get(vertex));
+			if (vertex.getId().equalsIgnoreCase("1")) {
+				Assert.assertEquals(OnmsSeverity.MAJOR.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+			} else if (vertex.getId().equalsIgnoreCase("2")) {
+				Assert.assertEquals(OnmsSeverity.MINOR.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+			} else if (vertex.getId().equalsIgnoreCase("3")) {
+				Assert.assertEquals(OnmsSeverity.MINOR.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+			} else if (vertex.getId().equalsIgnoreCase("4")) {
+				Assert.assertEquals(OnmsSeverity.MAJOR.getLabel().toLowerCase(), stats.get(vertex).computeStatus().toLowerCase());
+			}
+		}
+	}
+
+	/**
+	 * This method uses {@link PathOutageStatusProvider} object to retrieve status information for the visible vertices
+	 * from the test database
+	 * @return Map with {@link Status} information for all vertices
+	 */
+	private Map<VertexRef, Status> testStatuses() {
+		Map<VertexRef, Status> retvals = this.pathOutageStatusProvider.getStatusForVertices(pathOutageProvider,
```

Comment:
The `StatusProvider` does not use any criteria to calculate the status. There is no need to add criteria, simply omit the paramter.